tasks/post-to-cmr/index.js

```diff
@@ -78,7 +79,15 @@ async function postToCMR(event) {
   granules.forEach((granule) => {
     allGranules[granule.granuleId] = granule;
     granule.files.forEach((file) => {
-      allFiles.push(file.filename);
+      if (file.bucket && file.filepath) {
+        allFiles.push(buildS3Uri(file.bucket, file.filepath));
+      }
+      else if (file.filename) {
```

Comment:
It's confusing.  If `granule.files` no longer has `filename`, why do we still check `file.bucket && file.filepath` and `file.filename`?  Should `file.bucket && file.filepath` always present?

Reply:
It depends on where the input to the `post-to-cmr` task is coming from.  If it's coming from `discover-granules` (or any other task), it's going to have a `filename` field.  If the granules being used as input came from the API then they will have `bucket` and `filepath` fields, but no `filename`.